### ISSUE TITLE
Spike - Investigate options for updating a11y props being required or not

### ISSUE THREAD

**Issue by thatblindgeye** [patternfly/patternfly-react#9816]
**Describe the enhancement or change**
Spike for a discussion to have regarding https://github.com/patternfly/patternfly-react/issues/5397

Basically we need to decide whether:

- we want to update certain a11y props to either be required (if currently optional) or make them optional (if currently required)
- we want to add console warnings when an optional a11y prop is not passed in, but ideally should be (e.g. if an element doesn't have a visible text label, typically we should want an `aria-label` added somewhere)
- leave things as-is - props will remain optional or required as they already are
- some combination of all of the above

**Is this request originating from a Red Hat product team? If so, which ones and is there any sort of deadline for this enhancement?**

**Any other information?**
